### ISSUE TITLE
no-duplicate-id false positive

### ISSUE THREAD

**Issue by boris-petrov** [ember-template-lint/ember-template-lint#1511]
```hbs
{{#if this.foo}}
  <div id="foobar"></div>
{{else}}
  <span id="foobar"></span>
{{/if}}
```

This gives a warning on the `span` element but it shouldn't. Not sure how easy it is to be fixed though.

**Comment by MelSumner** [comment]
This is showing duplicate ID values- what part is a false positive to you? Can you say more?

**Comment by boris-petrov** [comment]
Well, because of the if-clause, always only one of the `div` or the `span` will be rendered - hence, there won't be a duplicate ID ever. So a warning shouldn't be triggered by `ember-template-lint`.

**Comment by MelSumner** [comment]
oh I see what you mean. Yes, we can fix for this. Thanks for pointing it out. 👍🏻

**Comment by josephdsumner** [comment]
@boris-petrov I can devote some time to trying to find a solution for this. I am familiar with the rule, but I'm not sure off the top of my head how easy this would be to track comprehensively. To double-check, the primary and non-negotiable assumption for your use case / similar cases is that only one of those `id` assignments will ever happen, correct? 

In the meantime, do you have any other examples of false positives that I can leverage for testing purposes?

**Comment by MelSumner** [comment]
@josephdsumner sure sounds good- 

The test that would go in the "good" list- 

```
    // BlockStatement if/else
    '{{#if this.foo}}<div id="foobar"></div>{{else}}<span id="foobar"></span>{{/if}}',
```

- we'd want to check duplicate ids and disregard them if they are in both the `{{#if}}` and the `{{else}}`
- we still want to check within the {{#if}} for duplicates, and also the `{{else}}` for duplicates

Does that make sense?

**Comment by boris-petrov** [comment]
@josephdsumner - yes, the assumption is that only one of the two HTML elements will be rendered at each and every time so that's why there is no need for the warning.

My use-case was very similar to the one I've put in the first comment. Only thing to note is that `{{else if this.bar}}` should also be handled, so you might put that in the test case @MelSumner.

**Comment by lifeart** [comment]
it's quite hard to solve all possible cases like this, especially for named blocks usage and component blocks itself.

I think it may be safely rewitten as 

```hbs
{{#let 'foobar' as |footerId|}}

{{#if this.foo}}
  <div id={{footerId}}></div>
{{else}}
  <span id={{footerId}}></span>
{{/if}}

{{/let}}
```

**Comment by boris-petrov** [comment]
@lifeart - yes, the general case is undecidable of course, but for the simple case like in my original post it should be doable. And I guess that will be the vast majority of cases.

**Comment by rwjblue** [comment]
Ya, I think any targeted fix here would be limited to static `id` values on either side of an `if`/`else`/`else if`. 

The following would need to be added as "Good" examples:

```hbs
{{#if this.whatever}}
  <div id="some static thing"></div>
{{else}}
  <div id="some static thing"></div>
{{/if}}
```

```hbs
{{#if this.whatever}}
  <div id="some static thing"></div>
{{else if this.otherThing}}
  <div id="some static thing"></div>
{{else}}
  <div id="some static thing"></div>
{{/if}}
```

(yes this is "icky", but technically still a language feature even though its forbidden by other rules)
```hbs
{{#unless this.whatever}}
  <div id="some static thing"></div>
{{else}}
  <div id="some static thing"></div>
{{/if}}
```

```hbs
{{#unless this.whatever}}
  <div id="some static thing"></div>
{{else unless this.otherThing}}
  <div id="some static thing"></div>
{{else if this.lolWtfBBQ}}
  <div id="some static thing"></div>
{{else}}
  <div id="some static thing"></div>
{{/if}}
```

**Comment by lifeart** [comment]
for `else if` case we could get location highlighting issues, due to https://github.com/glimmerjs/glimmer-vm/issues/1091
